src/Core/Tools/Authorizer/PostAuthorizer.php

```diff
@@ -139,13 +143,21 @@ public function isAllowed(Entity $entity, $privilege)
         }
 
         // If the user is the owner of this post & they have edit own posts permission
-        // they are allowed to edit or delete the post. They can't change the post status or
+        // they are allowed to edit the post. They can't change the post status or
         // ownership but those are already checked above
         if ($this->isUserOwner($entity, $user)
-            && in_array($privilege, ['update', 'delete', 'lock'])
+            && in_array($privilege, ['update', 'lock'])
             && $this->acl->hasPermission($user, Permission::EDIT_OWN_POSTS)) {
             return true;
         }
+        
+        // If the user is the owner of this post & they have delete own posts permission
+        // they are allowed to edit or delete the post.
+        if ($this->isUserOwner($entity, $user)
```

Comment:
It's a small nitpick, but for readability sake I believe it would be beneficial to just move this if condition to separate method.

I was thinking about something like:

```php
private function canUserDelete(Model $entity, User $user, string $privilege): bool
{
    return $this->isUserOwner($entity, $user)
        && $privilege === 'delete'
        && $this->acl->hasPermission($user, Permission::DELETE_OWN_POSTS);
}
```

And here just check for:

```php
if ($this->canUserDelete($entity, $user, $privilege)) {
    return true;
}

Reply:
I totally agree with you,  this file has many if conditions like this one 
later may we can refactor all of them together to make the code more readable